internal/controllers/successors.go

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import (
+	"fmt"
+
+	mmsemver "github.com/Masterminds/semver/v3"
+
+	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
+	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
+)
+
+// successorsPredicateFunc returns a predicate to find successors
+// for a bundle. Predicate must not include the current version.
+type successorsPredicateFunc func(bundle *catalogmetadata.Bundle) (catalogfilter.Predicate[catalogmetadata.Bundle], error)
```

Comment:
Can we also have tests for this? Probably a follow up is also good?

Reply:
Yes, we have tests cases in [here](https://github.com/operator-framework/operator-controller/blob/25af3b02ab8e2737ec982628dccdaca38f0dc618/internal/resolution/variablesources/installed_package_test.go). I'll have tests for the final version.

Needs a bit of work since I can't just copy-paste due to changed changes to the code for new context + these tests were not direct (they were testing `variablesources.MakeInstalledPackageVariables`). But for sure I want to cover this.